File: local.bc/goldcut/framework/prototype.php
<?php
class Prototype {

    private $inDomain;
    private $ofClass;
    private $ofType;

    public function __construct($str)
    {
        if (!$str) throw new Exception("NULL PROTOTYPE");
        $a = explode(':', $str);
        $this->inDomain = $a[0];
        $this->ofClass = $a[1];
        $this->ofType = $a[2];
    }

    public function __toString()
    {
        return "{$this->inDomain}:{$this->ofClass}:{$this->ofType}";
    }

    public function toTableName()
    {
        return "{$this->inDomain}_{$this->ofClass}_{$this->ofType}";
    }

    public function toColumnName()
    {
        return "{$this->inDomain}{$this->ofClass}{$this->ofType}";
    }

    /**
     * @return mixed
     */
    public function getInDomain()
    {
        return $this->inDomain;
    }

    /**
     * @param mixed $inDomain
     */
    public function setInDomain($inDomain)
    {
        $this->inDomain = $inDomain;
    }

    /**
     * @return mixed
     */
    public function getOfClass()
    {
        return $this->ofClass;
    }

    /**
     * @param mixed $ofClass
     */
    public function setOfClass($ofClass)
    {
        $this->ofClass = $ofClass;
    }

    /**
     * @return mixed
     */
    public function getOfType()
    {
        return $this->ofType;
    }

    /**
     * @param mixed $ofType
     */
    public function setOfType($ofType)
    {
        $this->ofType = $ofType;
    }



}
?>

File: local.bc/BCJava/src/main/java/digital/erp/symbol/Prototype.java
package digital.erp.symbol;

public class Prototype {

    private String inDomain;
    private String ofClass;
    private String ofType;

    public String getInDomain() {
        return inDomain;
    }

    public void setInDomain(String inDomain) {
        this.inDomain = inDomain;
    }

    public String getOfClass() {
        return ofClass;
    }

    public void setOfClass(String ofClass) {
        this.ofClass = ofClass;
    }

    public String getOfType() {
        return ofType;
    }

    public void setOfType(String ofType) {
        this.ofType = ofType;
    }

    public Prototype(String inDomain, String ofClass, String ofType)
    {
        this.inDomain = inDomain;
        this.ofClass = ofClass;
        this.ofType = ofType;
    }

    public String toString()
    {
        return this.inDomain + ":" + this.ofClass + ":" + this.ofType;
    }

    public String getAlias()
    {
        return this.inDomain + "_" + this.ofClass + "_" + this.ofType;
    }

    public String getShortName()
    {
        return this.inDomain + "" + this.ofClass + "" + this.ofType;
    }

    public static Prototype fromString(String protostring) throws PrototypeException.IncorrectFormat
    {
        String[] urna = protostring.split(":");
        if (urna.length == 3) {
            return new Prototype(urna[0], urna[1], urna[2]);
        }
        else
            throw new PrototypeException.IncorrectFormat(protostring);
    }
}
